tests/test_cmdline.py

```diff
@@ -0,0 +1,138 @@
+import zipline.__main__ as main
+import zipline
+from zipline.testing import ZiplineTestCase
+from zipline.testing.predicates import (
+    assert_equal,
+    assert_raises_str,
+)
+from click.testing import CliRunner
+from zipline.extensions import (
+    Namespace,
+    create_args,
+    parse_extension_arg,
+)
+
+
+class CmdLineTestCase(ZiplineTestCase):
+
+    def init_instance_fixtures(self):
+        super(CmdLineTestCase, self).init_instance_fixtures()
+
+    def test_parse_args(self):
+        n = Namespace()
+
+        arg_dict = {}
+
+        arg_list = [
+            'key=value',
+            'arg1=test1',
+            'arg2=test2',
+            'arg_3=test3',
+            '_arg_4_=test4',
+        ]
+        for arg in arg_list:
+            parse_extension_arg(arg, arg_dict)
+        assert_equal(
+            arg_dict,
+            {
+                '_arg_4_': 'test4',
+                'arg_3': 'test3',
+                'arg2': 'test2',
+                'arg1': 'test1',
+                'key': 'value',
+            }
+        )
+        create_args(arg_list, n)
+        assert_equal(n.key, 'value')
+        assert_equal(n.arg1, 'test1')
+        assert_equal(n.arg2, 'test2')
+        assert_equal(n.arg_3, 'test3')
+        assert_equal(n._arg_4_, 'test4')
+
+        msg = (
+            'invalid extension argument 1=test3, '
+            'must be in key=value form'
+        )
+        with assert_raises_str(ValueError, msg):
+            parse_extension_arg('1=test3', {})
+        msg = (
+            'invalid extension argument arg4 test4, '
```

Comment:
should we quote the name of the key in this error message to make it clear that it parsed as a single string with a space?